core/src/main/java/com/linecorp/armeria/client/proxy/ProxyConfig.java

```diff
@@ -119,6 +119,21 @@ public static ConnectProxyConfig connect(
                                       requireNonNull(password, "password"), useTls);
     }
 
+    /**
+     * Creates a {@code ProxyConfig} configuration for HAProxy protocol.
+     *
+     * @param sourceAddress the source address
+     * @param destinationAddress the destination address
+     */
+    public static HAProxyConfig haproxy(
+            InetSocketAddress sourceAddress, InetSocketAddress destinationAddress) {
```

Comment:
It's fine to add another version which only takes `destinationAddress`?

Reply:
Will address after APIs discussion is finalized 😅